merge.go

```diff
@@ -321,17 +321,21 @@ type MergeFileInput struct {
 }
 
 // populate a C struct with merge file input, make sure to use freeMergeFileInput to clean up allocs
-func populateCMergeFileInput(c *C.git_merge_file_input, input MergeFileInput) {
+func populateCMergeFileInput(c *C.git_merge_file_input, input MergeFileInput) *C.char {
 	c.path = C.CString(input.Path)
+	var toFree *C.char
 	if input.Contents != nil {
-		c.ptr = (*C.char)(unsafe.Pointer(&input.Contents[0]))
+		toFree = C.CString(string(input.Contents))
+		c.ptr = toFree
```

Comment:
This makes us allocate an extra buffer. We also don't actually know that what we have is a string. It's some buffer, so making it go through the string type seems wrong.

If there's no way to work around this in Go, I'd rather we write some C so we're passing `&input.Contents[0]` to a C function so the rules make it OK for us to re-use this buffer.